dds/CMakeLists.txt

```diff
@@ -0,0 +1,635 @@
+    DCPS/transport/framework/TransportReceiveStrategy_T.inl
+    DCPS/transport/framework/TransportReceiveStrategy_rch.h
+    DCPS/transport/framework/TransportRegistry.h
+    DCPS/transport/framework/TransportRegistry.inl
+    DCPS/transport/framework/TransportReplacedElement.h
+    DCPS/transport/framework/TransportReplacedElement.inl
+    DCPS/transport/framework/TransportRetainedElement.h
+    DCPS/transport/framework/TransportRetainedElement.inl
+    DCPS/transport/framework/TransportSendBuffer.h
+    DCPS/transport/framework/TransportSendBuffer.inl
+    DCPS/transport/framework/TransportSendControlElement.h
+    DCPS/transport/framework/TransportSendControlElement.inl
+    DCPS/transport/framework/TransportSendElement.h
+    DCPS/transport/framework/TransportSendElement.inl
+    DCPS/transport/framework/TransportSendListener.h
+    DCPS/transport/framework/TransportSendStrategy.h
+    DCPS/transport/framework/TransportSendStrategy.inl
+    DCPS/transport/framework/TransportSendStrategy_rch.h
+    DCPS/transport/framework/TransportStatistics.h
+    DCPS/transport/framework/TransportStrategy.h
+    DCPS/transport/framework/TransportStrategy_rch.h
+    DCPS/transport/framework/TransportType.h
+    DCPS/transport/framework/TransportType_rch.h
+    DCPS/unique_ptr.h
+    DCPS/yard/yard_base_grammar.hpp
+    DCPS/yard/yard_char_set.hpp
+    DCPS/yard/yard_parser.hpp
+    DCPS/yard/yard_text_grammar.hpp
+    DCPS/yard/yard_tree.hpp
+    Version.h
+    Versioned_Namespace.h
+)
+target_compile_definitions(OpenDDS_Dcps PUBLIC ${OPENDDS_DCPS_COMPILE_DEFINITIONS})
+_opendds_library(OpenDDS_Dcps MSVC_BIGOBJ)
+target_include_directories(OpenDDS_Dcps
+  PUBLIC
+    "$<BUILD_INTERFACE:${OPENDDS_SOURCE_DIR}>"
+    "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>"
+  PRIVATE
+    "${CMAKE_CURRENT_SOURCE_DIR}"
+)
+if(OPENDDS_RAPIDJSON)
+  target_include_directories(OpenDDS_Dcps PUBLIC "$<BUILD_INTERFACE:${OPENDDS_RAPIDJSON}/include>")
+endif()
+target_link_libraries(OpenDDS_Dcps PUBLIC ${dcps_lib_deps})
+
+set(opendds_target_sources_args
+  AUTO_LINK FALSE
+  USE_EXPORT "dds/DCPS/dcps_export.h" OpenDDS_Dcps_Export
+  INCLUDE_BASE "${OPENDDS_SOURCE_DIR}"
+  USE_VERSIONED_NAMESPACE "dds/Versioned_Namespace.h" OPENDDS
+)
+
+opendds_target_sources(OpenDDS_Dcps
+  PUBLIC
+    CorbaSeq/BooleanSeq.idl
+    CorbaSeq/CharSeq.idl
+    CorbaSeq/DoubleSeq.idl
+    CorbaSeq/FloatSeq.idl
+    CorbaSeq/Int8Seq.idl
+    CorbaSeq/LongDoubleSeq.idl
+    CorbaSeq/LongLongSeq.idl
+    CorbaSeq/LongSeq.idl
+    CorbaSeq/OctetSeq.idl
+    CorbaSeq/ShortSeq.idl
+    CorbaSeq/StringSeq.idl
+    CorbaSeq/UInt8Seq.idl
+    CorbaSeq/ULongLongSeq.idl
+    CorbaSeq/ULongSeq.idl
+    CorbaSeq/UShortSeq.idl
+    CorbaSeq/WCharSeq.idl
+    CorbaSeq/WStringSeq.idl
+  SKIP_TAO_IDL
+  OPENDDS_IDL_OPTIONS -SI -Wb,tao_include_prefix=tao/
+  ${opendds_target_sources_args}
+)
+
+opendds_target_sources(OpenDDS_Dcps
+  PUBLIC
+    DdsDcps.idl
+    DdsDcpsConditionSeq.idl # TODO
```

Comment:
TODOs?

Reply:
I think these are safety profile TODOs. I will double check this and note that if that's the case.